need to explicitely include <optional>

Hi,

I am not able to build opustags as it fails during compilation because of

```
In file included from /home/op/build/opustags/src/cli.cc:10:
/home/op/build/opustags/src/opustags.h:390:7: error: no template named
      'optional' in namespace 'std'
        std::optional<std::string> path_out;
        ~~~~~^
...
```

I am not proficient in C++, so I'm not able to properly fix this, but if I add an explicit include in opustags, it builds fine.

```diff
Index: src/opustags.h
--- src/opustags.h.orig
+++ src/opustags.h
@@ -31,6 +31,7 @@
 #include <functional>
 #include <list>
 #include <memory>
+#include <optional>
 #include <string>
 #include <vector>
 ```

I'm using clang 10.0.1 on OpenBSD.  Also, FYI to build here it also needs the `target_link_libraries` to iconv.

I've just fixed a bunch of tags on various ogg files, thanks for developing this :)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,8 +19,10 @@ pkg_check_modules(OGG REQUIRED ogg)
 add_compile_options(${OGG_CFLAGS})
 link_directories(${OGG_LIBRARY_DIRS})
 
+include(FindIconv)
+
 configure_file(src/config.h.in config.h @ONLY)
-include_directories(BEFORE src "${CMAKE_BINARY_DIR}" ${OGG_INCLUDE_DIRS})
+include_directories(BEFORE src "${CMAKE_BINARY_DIR}" ${OGG_INCLUDE_DIRS} ${Iconv_INCLUDE_DIRS})
 
 add_library(
 	ot
@@ -30,11 +32,7 @@ add_library(
 	src/opus.cc
 	src/system.cc
 )
-target_link_libraries(ot PUBLIC ${OGG_LIBRARIES})
-
-if (APPLE)
-	target_link_libraries(ot PUBLIC iconv)
-endif()
+target_link_libraries(ot PUBLIC ${OGG_LIBRARIES} ${Iconv_LIBRARIES})
 
 add_executable(opustags src/opustags.cc)
 target_link_libraries(opustags ot)

